docs/_Interface_IColonyNetwork.md

```diff
@@ -0,0 +1,858 @@
+---
+title: IColonyNetwork
+section: Interface
+order: 2
+---
+
+  
+
+## Interface Methods
+
+
+
+### `addColonyVersion`
+
+Adds a new Colony contract version and the address of associated `_resolver` contract. Secured function to authorised members
+
+
+
+**Parameters**
+
+
+|Name|Type|Description|
+|---|---|---|
+|_version|uint256| The new Colony contract version
+|_resolver|address| Address of the `Resolver` contract which will be used with the underlying `EtherRouter` contract
+
+
+
+
+
+### `addr`
+
+Returns the address the supplied node resolves do, if we are the resolver
+
+
+
+**Parameters**
+
+
+|Name|Type|Description|
+|---|---|---|
+|node|bytes32| The namehash of the ENS address being requested
+
+
+
+**Return Parameters**
+
+
+|Name|Type|Description|
+|---|---|---|
+|_undefined_|address|
```

Comment:
Badly parsed return parameter

Reply:
What would you rather have here if there is no name for the variable?

Reply:
I would rather have the description from natspec i.e. `The address the supplied node resolves to`

Reply:
```
function addr(bytes32 node) public view returns (address);
```

Reply:
Currently it checks the name of the parameter. I can make it check the type if there is no name provided (in this case it would use `address`) or we can add a name to the return parameters that are missing a name.

Reply:
```
   /// @notice Returns the address the supplied node resolves do, if we are the resolver
  /// @param node The namehash of the ENS address being requested
  /// @return address The address the supplied node resolves to
  function addr(bytes32 node) public view returns (address);
```

Reply:
I was questioning the return parameter description rather than name or type.

Reply:
In order to verify that the description is the right description for the given param, it checks the first word in the description (currently the name of the param).

Reply:
I can make it check the type if no name is provided. It would solve this particular case.

Reply:
We can add names to all return parameters. Most should be there already so this is more of an exception

Reply:
I think adding names to the return parameters would be nice. It falls back to the type if no name is provided. That resolved this specific case and one other point you pointed out.